File: src/main/java/com/github/zoewithabang/service/OptionService.java
package com.github.zoewithabang.service;

import com.github.zoewithabang.dao.OptionDao;

import java.awt.*;
import java.sql.Connection;
import java.sql.SQLException;
import java.util.Properties;

public class OptionService implements IService
{
    private Properties botProperties;
    private String database;
    private OptionDao optionDao;
    
    public OptionService(Properties botProperties)
    {
        this.botProperties = botProperties;
        database = botProperties.getProperty("dbdatabase");
        optionDao = new OptionDao(botProperties);
    }
    
    public String getOptionValue(String key) throws SQLException
    {
        try(Connection connection = optionDao.getConnection(database))
        {
            return optionDao.get(connection, key).getValue();
        }
        catch(SQLException e)
        {
            LOGGER.error("SQLException on getting Option value for key '{}'.", key, e);
            throw e;
        }
    }
    
    public Color getBotColour()
    {
        try(Connection connection = optionDao.getConnection(database))
        {
            return Color.decode(optionDao.get(connection, "colour").getValue());
        }
        catch(SQLException | NumberFormatException e)
        {
            LOGGER.warn("Exception on getting bot colour, defaulting to black");
            return Color.BLACK;
        }
    }
}

File: src/main/kotlin/com/github/zoewithabang/service/OptionService.kt
package com.github.zoewithabang.service

import com.github.zoewithabang.dao.OptionDao
import org.apache.logging.log4j.LogManager

import java.awt.*
import java.sql.Connection
import java.sql.SQLException
import java.util.Properties

class OptionService(private val botProperties: Properties) {
    private val logger = LogManager.getLogger("OptionService")
    private val database: String = botProperties.getProperty("dbdatabase")
    private val optionDao: OptionDao = OptionDao(botProperties)

    val botColour: Color
        get() {
            try {
                optionDao.getConnection(database)
                    .use { connection -> return Color.decode(optionDao[connection, "colour"]!!.value) }
            } catch (e: SQLException) {
                logger.warn("Exception on getting bot colour, defaulting to black")

                return Color.BLACK
            } catch (e: NumberFormatException) {
                logger.warn("Exception on getting bot colour, defaulting to black")

                return Color.BLACK
            }
        }

    @Throws(SQLException::class)
    fun getOptionValue(key: String): String {
        try {
            optionDao.getConnection(database).use { connection -> return optionDao[connection, key]!!.value!! }
        } catch (e: SQLException) {
            logger.error("SQLException on getting Option value for key '{}'.", key, e)

            throw e
        }
    }
}
